README.md

```diff
@@ -1,285 +1,366 @@
 
-- **`buyAllAmountPayEth`** - Used to exchange `ETH` for `ERC-20 token`. This methods should be called with `ETH`. In the beginning the sent `ETH` is locked in `WETH token`. After that the `otc` contract is ensured to have enough allowance in the `wethToken` to do transfers on the behalf of `this` contract. Follows buy operation where `wethAmt` is calculated. The bought amount is transferred to the caller. Last step include sending the difference between the amount of `ETH` sent to the contract method and the actual amount needed to buy the `buyToken` amount.
-**Disclaimer:** It's important to pay attention that there is neither `minBuyAmt` nor `maxPayAmt`. The caller of the method should take into account the slippage that might occur and send more `ETH` ( the max amount he is willing to pay in order to get the `buyAmt`. If less `ETH` is spent, the remaining is returned to the caller.
+The value provided here determines the lowest possible price that allows the trade to go through and close. If the prices drop below the threshold, the transaction fails and everything is reverted. Following the example above, one can provide a value (in base units) such as `140 DAI`. This will allow the transaction to pass and then the trade will be deemed successful if the price of `MKR/DAI` does not drop below `1.4 MKR/DAI`.
 
-	**Arguments:**
-	- _[OtcInterface](https://github.com/makerdao/maker-otc-support-methods/blob/master/src/MakerOtcSupportMethods.sol#L5) `otc`_ - An address of the contract that represents the OTC market contract. Only a few methods are exposed.
-	 - _[TokenInterface](https://github.com/makerdao/oasis-direct-proxy/blob/master/src/OasisDirectProxy.sol#L11) `buyToken`_ - An address of any `ERC-20 token` or token that implements the interface
-	 - _`uint buyAmt`_ - The amount that will be bought
-	 - _[TokenInterface](https://github.com/makerdao/oasis-direct-proxy/blob/master/src/OasisDirectProxy.sol#L11) `wethToken`_ -The address of the `WETH token`
 
-	**Modifiers**
-	- _`*payable`_
+### Constructor
+**Summary:** This is used to inject the `WETH` token address which is used in some of the methods described below.
 
-	**Returns:**
-	- _`uint wethAmth`_ -the amount of `WETH token` that will be paid
+**Arguments**:
 
-	**Errors**
-	- _`otc.buyAllAmount`_ - might cause the TX to fail if the the caller is expected to pay more `ETH` than he sent.
-	- _`buyToken.transfer`_ - might cause the TX to fail
-	- _`withdrawAndSend`_ - might cause the TX to fail
+- `address wethToken` - The address of the [WETH](https://github.com/dapphub/ds-weth/blob/master/src/weth9.sol)  token. Note that it must be provided as a contract creation because the reference is needed in the `fallback` function.
```

Comment:
What does it mean as a contract creation? You have to pass the WETH token address only :)